rest-catalog/quarkus-rest-catalog/src/intTest/java/org/projectnessie/restcatalog/server/BaseNessieRestCatalogTests.java

```diff
@@ -0,0 +1,179 @@
+import org.projectnessie.restcatalog.server.resources.NessieDockerTestResourceLifecycleManager.NessieDockerUri;
+import org.projectnessie.restcatalog.server.resources.WarehouseTestLocation;
+import org.testcontainers.containers.Network;
+
+public abstract class BaseNessieRestCatalogTests extends CatalogTests<RESTCatalog> {
+
+  @NessieDockerUri URI nessieUri;
+  @KeycloakTokenEndpointUri URI oidcTokenEndpointUri;
+  @KeycloakClientId String oidcClientId;
+  @KeycloakClientSecret String oidcClientSecret;
+
+  @BeforeEach
+  public void clearRepo() throws Exception {
+    try (NessieApiV2 controlClient = buildControlClient()) {
+      Branch main = controlClient.getDefaultBranch();
+      Branch empty =
+          Branch.of(
+              main.getName(), "2e1cfa82b035c26cbbbdae632cea070514eb8b773f616aaeaf668e2f0be8f10d");
+
+      controlClient.getAllReferences().stream()
+          .forEach(
+              ref -> {
+                try {
+                  if (ref instanceof Branch) {
+                    if (!ref.getName().equals(main.getName())) {
+                      controlClient.deleteBranch().branch((Branch) ref).delete();
+                    }
+                  } else if (ref instanceof Tag) {
+                    controlClient.deleteTag().tag((Tag) ref).delete();
+                  }
+                } catch (NessieConflictException | NessieNotFoundException e) {
+                  throw new RuntimeException(e);
+                }
+              });
+
+      checkState(
+          controlClient.assignBranch().branch(main).assignTo(empty).assignAndGet().equals(empty));
+    }
+  }
+
+  @Override
+  protected boolean requiresNamespaceCreate() {
+    return true;
+  }
+
+  @Override
+  protected boolean supportsNamespaceProperties() {
+    return true;
+  }
+
+  @Override
+  protected boolean supportsServerSideRetry() {
+    return true;
+  }
+
+  @Override
+  protected boolean supportsNestedNamespaces() {
+    return true;
+  }
+
+  @Override
+  protected RESTCatalog catalog() {
+    try (NessieApiV2 controlClient = buildControlClient()) {
+      RESTCatalog catalog = new RESTCatalog();
+      catalog.setConf(new Configuration());
+      catalog.initialize(
+          "nessie-iceberg-rest",
+          ImmutableMap.of(
+              CatalogProperties.URI,
+              restCatalogServerUri(),
+              CatalogProperties.FILE_IO_IMPL,
+              "org.apache.iceberg.io.ResolvingFileIO",
+              OAuth2Properties.CREDENTIAL,
+              oidcClientId + ":" + oidcClientSecret,
+              OAuth2Properties.SCOPE,
+              "profile",
+              "prefix",
+              requireNonNull(controlClient.getConfig().getDefaultBranch())));
+      return catalog;
+    }
+  }
+
+  private String restCatalogServerUri() {
+    return format(
+        "http://127.0.0.1:%d/iceberg/",
+        Objects.requireNonNull(
+            Integer.getInteger("quarkus.http.test-port"),
```

Comment:
Hm. Where does this property come from, now that the apprunner-plugin's not used.

Reply:
It's a random port assigned by Quarkus. The property is declared in the `application.properties` file in this project:

```
quarkus.http.test-port=0
```

My understanding is that when the port is set to zero, a random port is assigned, then Quarkus sets the effective port as a system property for other components to consume it. See here:

https://github.com/quarkusio/quarkus/blob/main/test-framework/common/src/main/java/io/quarkus/test/common/LauncherUtil.java#L243

Reply:
I think we can simplify this. It's totally fine imho to use a fixed port, e.g. 19131. I will try.

Reply:
Nope. Fixed ports seem to pose conflict problems. Let's keep the random port.

Reply:
Ah, okay. That was something the apprunner-plugin did in the past. All good.

Reply:
Random port are better for testing IMO.